analysis/cross_partition_combiners.py

```diff
@@ -282,33 +278,59 @@ class CrossPartitionCombiner(pipeline_dp.combiners.Combiner):
     # 1. The sum of non dp metrics, which is used for averaging of error
     # metrics.
     # 2. metrics.UtilityReport contains error metrics.
-    AccumulatorType = Tuple[Tuple, metrics.UtilityReport]
-
-    def __init__(self, dp_metrics: List[pipeline_dp.Metrics],
-                 public_partitions: bool):
+    AccumulatorType = Tuple[Tuple, metrics.UtilityReport, Tuple[float]]
+
+    @staticmethod
+    def partition_contribution_weighing_function(
+            metrics: metrics.PerPartitionMetrics) -> Tuple[float]:
+        return tuple(map(lambda el: el.sum, metrics.metric_errors))
+
+    @staticmethod
+    def equal_weighing_function(
+            metrics: metrics.PerPartitionMetrics) -> Tuple[float]:
+        """
+        Assumes that partition_selection_probability_to_keep for public
+        partitions is 1 and all public partitions including are processed
+        via `create_accumulator`.
+        For the public partitions weights will be 1, and we will do normal
+        averaging because total weight will equal to the total number of
+        partitions. For private partitions we will do weighted average and
+        total weight will equal to mean number of kept partitions
+        (`partitions.kept_partitions.mean`).
+        """
+        return (metrics.partition_selection_probability_to_keep,) * len(
```

Comment:
I thought again, let for now for simplicity assume that weight is just a number (i.e. it's the same for all metrics)

Reply:
but will it really simplify the code? I think it won't look much simpler after it.